app/controllers/posts_controller.rb

```diff
@@ -4,6 +4,7 @@ class PostsController < ApplicationController
   # GET /posts
   # GET /posts.json
   def index
+    WeeklyEmail.perform_later
```

Comment:
This shouldn't be called from the controller, every time a user enters to this action the emails are going to be triggered, this should be called by the background job